codex/purchasing/states/unknown.nim

```diff
@@ -20,8 +22,8 @@ method enterAsync(state: PurchaseUnknown) {.async.} =
       of RequestState.Cancelled:
         state.switch(PurchaseCancelled())
       of RequestState.Finished:
-        discard # TODO
+        state.switch(PurchaseFinished())
       of RequestState.Failed:
-        discard # TODO
+        state.switch(PurchaseFailed())
   except CatchableError as error:
```

Comment:
Could we narrow the error type down a bit more? `JsonRpcProviderError` is an option, any others?